src/base/bittorrent/portforwarderimpl.cpp

```diff
@@ -63,30 +64,39 @@ void PortForwarderImpl::setEnabled(const bool enabled)
     m_storeActive = enabled;
 }
 
-void PortForwarderImpl::addPort(const quint16 port)
+void PortForwarderImpl::replacePorts(const QString &profile, QSet<quint16> ports)
 {
-    if (m_mappedPorts.contains(port))
-        return;
+    PortMapping &portMapping = m_portProfiles[profile];
+    Algorithm::removeIf(portMapping, [this, &ports](const quint16 port, const std::vector<lt::port_mapping_t> &handles)
```

Comment:
Always feel it confusing when such algorithms also mutate something other than target container. So generic way using loop would be more appropriate to represent the logic.

Reply:
>Always feel it confusing when such algorithms also mutate something other than target container. So generic way using loop would be more appropriate to represent the logic.

I don't really agree. The `Algorithm::removeIf` only take cares of the target container, that is, to make sure it loop over and remove elements without stepping on errors which is a common, error-prone *pattern*.
Other modifications is dealt by the lambda function which are free to do anything (as long as it don't modify the container). Most (if not all) of the higher-order functions are bounded by this rule.

Also I really doubt the new loop will be easier to maintain (maintaining iterators while removing some element is error prone!)